system_test/common/aest_compatibility_SUITE.erl

```diff
@@ -81,27 +84,51 @@ test_mining_algorithms_compatibility(Cfg) ->
     start_node(node2, Cfg),
     wait_for_startup([node1, node2], 0, Cfg),
 
-    wait_for_value({height, Length + 1}, [node1, node2],
-                   (Length + 1) * ?MINING_TIMEOUT, Cfg),
+    wait_for_value({height, MinLength + 1}, [node1, node2],
+                   (MinLength + 1) * ?MINING_TIMEOUT, Cfg),
+    wait_for_beneficiaries(node1, node2, [Beneficiary1, Beneficiary2],
+                           MinLength, ExtraStep, MaxSteps,
+                           Cfg),
+    ok.
+
+%=== INTERNAL FUNCTIONS ========================================================
 
-    B1 = get_block(node1, Length),
-    B2 = get_block(node2, Length),
+setup(NodeSpecs, Config, Cfg) ->
+    setup_nodes([maps:put(config, Config, N) || N <- NodeSpecs], Cfg).
 
+get_block_beneficiary(NodeName, Height) ->
+    #{ beneficiary := Beneficiary } = get_block(NodeName, Height),
+    Beneficiary.
+
+get_chain_beneficiaries(NodeName, Height) when is_integer(Height),
+                                               Height >= 1 ->
+    sets:to_list(
```

Comment:
```
lists:usort([ get_block_beneficiary(NodeName, H) || H <- lists:seq(1, Height) ])
```
is better?

Reply:
The set rather than full list was mainly for preventing copy-paste of function and reuse in case of many beneficiaries (avoiding creating long list).

Also, the direction of the traversals from top to bottom (still by height) is debatable.

It is good enough?

Reply:
Addressed in new commit, [passed CI](https://circleci.com/gh/aeternity/aeternity/43762).